documentation/project-docs/breaking-change-guidelines.md

```diff
@@ -0,0 +1,81 @@
+# SDK Breaking Change and Diagnostic Guidelines
+
+This document provides guidelines for introducing new diagnostics or breaking changes to the .NET SDK, which configuration knobs are available, and what criteria should guide the decision around severity, timeline for introduction, and deprecation of diagnostics.
+
+## Overall procedure
+
+In general, we want to make updating the .NET SDK as smooth as possible for developers. This means:
+
+* introducing new changes in a staged/gradual way
+* trying to tie opinionated analyzers/diagnostics to a mechanism that requires explicit user opt-in
```

Comment:
not sure what opinionated means in this context. Maybe just remove or are there specific types of diagnostics that you would include and some you wouldn't.

Reply:
Done.